mhs/mhs/outbound/request/synchronous/handler.py

```diff
@@ -1,91 +1,108 @@
-        """
+            interaction_details = self._get_interaction_details(interaction_id)
+        except common_workflow.UnknownInteractionError as e:
+            logger.warning('0007', 'Unknown {InteractionId} in request', {'InteractionId': interaction_id})
+            raise tornado.web.HTTPError(404, "Unknown interaction ID: %s", interaction_id) from e
+
         try:
-            logger.info('0003', 'Waiting for asynchronous response to message with {MessageId}',
-                        {'MessageId': async_message_id})
-            await self.async_response_received.wait(datetime.timedelta(seconds=self.async_timeout))
-        except TimeoutError:
-            raise tornado.web.HTTPError(log_message=f"Timed out waiting for a response to message {async_message_id}")
+            workflow = self.workflows[interaction_details['workflow']]
+        except KeyError as e:
+            logger.error('0008', "Weren't able to determine workflow for {InteractionId} . This likely is due to a "
+                                 "misconfiguration in interactions.json", {"InteractionId": interaction_id})
+            raise tornado.web.HTTPError(500, "Couldn't determine workflow to invoke for interaction ID: %s",
+                                        interaction_id) from e
+
+        status, response = await workflow.handle_supplier_message(message_id, interaction_details, body)
+
+        self._write_response(status, response)
+
+    def _extract_message_id(self):
+        message_id = self.request.headers.get('Message-Id', None)
+        if not message_id:
+            message_id = message_utilities.MessageUtilities.get_uuid()
+            log.message_id.set(message_id)
+            logger.info('0001', "Didn't receive message id in incoming request from supplier, so have generated a new "
+                                "one.")
+        else:
+            log.message_id.set(message_id)
+            logger.info('0002', 'Found message id on incoming request.')
+        return message_id
+
+    def _extract_correlation_id(self):
+        correlation_id = self.request.headers.get('Correlation-Id', None)
+        if not correlation_id:
+            correlation_id = message_utilities.MessageUtilities.get_uuid()
+            log.correlation_id.set(correlation_id)
+            logger.info('0003', "Didn't receive correlation id in incoming request from supplier, so have generated a "
+                                "new one.")
+        else:
+            log.correlation_id.set(correlation_id)
+            logger.info('0004', 'Found correlation id on incoming request.')
+
+    def _extract_interaction_id(self):
+        try:
+            interaction_id = self.request.headers['Interaction-Id']
+        except KeyError as e:
+            logger.warning('0005', 'Required Interaction-Id header not passed in request')
+            raise tornado.web.HTTPError(404, 'Required Interaction-Id header not found') from e
```

Comment:
Don't know if I'm on the right track with this but it feels to me like this method shouldn't know/care about tornado errors. Also if this method doesn't have a tornado error it might make extracting a common module from these methods a bit simpler if we ever do implement the outbound async queue handler.